platform/posix/transport/src/openssl_posix.c

```diff
@@ -792,47 +825,70 @@ int32_t Openssl_Send( NetworkContext_t * pNetworkContext,
 {
     OpensslParams_t * pOpensslParams = NULL;
     int32_t bytesSent = 0;
-    int32_t sslError = 0;
-
-    /* Unused parameter when logs are disabled. */
-    ( void ) sslError;
 
     if( ( pNetworkContext == NULL ) || ( pNetworkContext->pParams == NULL ) )
     {
         LogError( ( "Parameter check failed: pNetworkContext is NULL." ) );
     }
     else if( pNetworkContext->pParams->pSsl != NULL )
     {
+        struct pollfd pollFds;
+        int32_t pollStatus;
+
         pOpensslParams = pNetworkContext->pParams;
-        /* SSL write of data. */
-        bytesSent = ( int32_t ) SSL_write( pOpensslParams->pSsl,
-                                           pBuffer,
-                                           ( int32_t ) bytesToSend );
 
-        if( bytesSent <= 0 )
+        /* Initialize the file descriptor. */
+        pollFds.events = POLLOUT | POLLPRI | POLLHUP | POLLERR;
```

Comment:
Same comments here as [mentioned earlier](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1665/files?file-filters%5B%5D=.c#r669855972) about not needing `POLLHUP`, `POLLERR` flags to be explicitly set here as these are always enabled